logger.go

```diff
@@ -30,5 +68,11 @@ func (l *Logger) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.Ha
 	next(rw, r)
 
 	res := rw.(ResponseWriter)
-	l.Printf("%v | %v | \t %v | %v | %v %v", start.Format(time.RFC3339), res.Status(), time.Since(start), r.Host, r.Method, r.URL.Path)
+	log := Log{start.Format(l.dateFormat), res.Status(), time.Since(start), r.Host, r.Method, r.URL.Path}
+
+	l.mu.Lock()
+	defer l.mu.Unlock()
```

Comment:
Is the mutex necessary?  I was under the impression that the `text/template.Template.Execute` func was stateless.  It'd be cool if we could avoid an explicit per-request bottleneck.

If the mutex is for the buffer instead, then I think the buffer should be moved off of the logger given that it's only used within the body of this func.  Is the reuse of buffer intended to reduce GC churn?